### PR TITLE
Add-remove nuimo feature

### PR DESCRIPTION
**Pull request by commento**
- Add add-remove nuimos feature with nuimo_app restart
- solves issue #251 
- code for configuration-watcher for add-remove nuimos is present but not yet working
- Add nuimo's name in setup_app (name modification to be added)
- solve issue in case nuimo_app active component is removed

### CODE REVIEW

File: senic_hub/nuimo_app/__main__.py [hunk 1]
```diff
@@ -40,64 +41,88 @@ def main(config):
...
-            nuimo_apps[mac_addr].put({'method': 'stop'})
+    # TODO: remove nuimos without restart nuimo_app code, still not working
+    # for mac_addr in nuimo_apps.keys():
+    #     if mac_addr not in updated_nuimos:
+    #         logger.info("A Nuimo was removed from the config file: %s", mac_addr)
+    #         nuimo_apps[mac_addr].put({'method': 'stop'})
+    #         processes[mac_addr].join()
+    #         del nuimo_apps[mac_addr]
+    #         logger.info("PROCESS join %s mac addr %s join", processes[mac_addr], mac_addr)
+    #         del processes[mac_addr]
```
Comment (rudolfs) [inline senic_hub/nuimo_app/__main__.py]: Remove commented code -> create a new issue.

File: senic_hub/nuimo_app/__main__.py [hunk 2]
```diff
@@ -40,64 +41,88 @@ def main(config):
...
-            Process(target=app.start, args=(ipc_queue,)).start()
+        # TODO: add nuimos without restart nuimo_app code, still not working
+        # else:
+        #     logger.info("A new Nuimo was found in the config file: %s", mac_addr)
+        #     app = NuimoApp(ha_api_url, ble_adapter_name,
+        #              mac_addr, components)
+        #     ipc_queue = Queue()
+        #     nuimo_apps[mac_addr] = ipc_queue
+        #     processes[mac_addr] = Process(target=app.start, args=(ipc_queue,))
+        #     processes[mac_addr].start()
```
Comment (rudolfs) [inline senic_hub/nuimo_app/__main__.py]: Ditto.

File: senic_hub/nuimo_app/__main__.py [hunk 3]
```diff
@@ -24,10 +25,10 @@
 @click.option('--config', '-c', required=True, type=click.Path(exists=True), help="app configuration file")
 def main(config):
     log_format = '%(asctime)s %(levelname)-5.5s [%(name)s] %(message)s'
-    logging.basicConfig(level=logging.DEBUG, format=log_format)
+    logging.basicConfig(level=logging.INFO, format=log_format)
```
Comment (commento) [inline senic_hub/nuimo_app/__main__.py]: are you serious?

Comment (rudolfs) [inline senic_hub/nuimo_app/__main__.py]: This comment was from before we talked...

File: senic_hub/nuimo_app/__init__.py [hunk 4]
```diff
@@ -240,13 +250,16 @@ def listen_to_ipc_queue(self, ipc_queue):
...
-                logger.debug("IPC set_components() received: %s", components)
+                logger.info("IPC set_components() received: %s mac = %s", components, self.controller.mac_address)
                 component_instances = get_component_instances(components)
                 self.set_components(component_instances)
             elif msg['method'] == 'stop':
-                logger.debug("IPC stop() received")
+                logger.info("IPC stop() received %s", self.controller.mac_address)
                 self.stop()
-                return
+                break
```
Comment (commento) [inline senic_hub/nuimo_app/__init__.py]: `break` is used to break a loop (in this case infinite loop)

File: senic_hub/nuimo_app/__init__.py [hunk 5]
```diff
@@ -240,13 +250,16 @@ def listen_to_ipc_queue(self, ipc_queue):
...
+
+        logger.info("the work for the Thread is finished, return")
```
Comment (commento) [inline senic_hub/nuimo_app/__init__.py]: if `break` returns `return` this trace will be removed

File: senic_hub/nuimo_app/__init__.py [hunk 6]
```diff
@@ -15,11 +15,11 @@ class NuimoControllerListener(ControllerListener):
 
     def started_connecting(self):
         mac = self.controller.mac_address
-        logger.info("Connecting to Nuimo controller %s...", mac)
+        logger.info("started_connecting to Nuimo controller %s...", mac)
 
     def connect_succeeded(self):
         mac = self.controller.mac_address
-        logger.info("Connected to Nuimo controller %s", mac)
+        logger.info("connect_succeeded to Nuimo controller %s", mac)
```
Comment (commento) [inline senic_hub/nuimo_app/__init__.py]: Touché

File: senic_hub/nuimo_app/__init__.py [hunk 7]
```diff
@@ -104,7 +112,9 @@ def stop(self):
             self.active_component.stop()
 
         self.controller.disconnect()
+        logger.info("Disconnected %s", self.controller.mac_address)
```
Comment (commento) [inline senic_hub/nuimo_app/__init__.py]: 39a0407f00032361fa7f15510828741b2d0ad138

**Pull request merged**